modules/pam_motd/pam_motd.c

```diff
@@ -215,6 +209,47 @@ static void try_to_display_directories_with_overrides(pam_handle_t *pamh,
     if (dirscans_size_total == 0)
         goto out;
 
+    /* filter out unwanted names, directories, and complement data with lstat() */
+    for (i = 0; i < num_motd_dirs; i++) {
+	unsigned int j;
+	int rc, drop;
+	struct dirent **d;
+	char *fullpath;
+	struct stat s;
```

Comment:
Since most of these variables are used inside the inner loop over `j` variable,
it would be better to move them inside that loop.

Reply:
Okay, agree.